File: selfstudy/golang/cities_avg.go
/*
task 274:
Cities - Avg
You need to compute the average of city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func AverageCities(populations []int) float64 {
	if len(populations) == 0 { return 0 }
	sum := 0
	for _, v := range populations {
		sum += v
	}
	return float64(sum)/float64(len(populations))
}

func main() {
	tests := []struct {
		input []int
		expected float64
	}{
		{[]int{10, 20, 30}, AverageCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, AverageCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, AverageCities([]int{1, 2, 3})},
		{[]int{100, 200}, AverageCities([]int{100, 200})},
		{[]int{}, AverageCities([]int{})},
	}
	
	for idx, t := range tests {
		res := AverageCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/python/cities_avg.py
'''
task 174:
Cities - Avg
You need to compute the average of city populations in thousands
'''

def average_cities(populations):
    return sum(populations) / len(populations) if populations else 0

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], average_cities([10, 20, 30])),
        ([50, 60, 70], average_cities([50, 60, 70])),
        ([1, 2, 3], average_cities([1, 2, 3])),
        ([100, 200], average_cities([100, 200])),
        ([], average_cities([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = average_cities(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think
